packages/http-server/src/http-server.ts

```diff
@@ -3,80 +3,32 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {IncomingMessage, ServerResponse} from 'http';
 import * as http from 'http';
 import * as https from 'https';
 import {AddressInfo} from 'net';
-
 import * as pEvent from 'p-event';
-
-export type RequestListener = (
-  req: IncomingMessage,
-  res: ServerResponse,
-) => void;
-
-/**
- * Basic HTTP server listener options
- *
- * @export
- * @interface ListenerOptions
- */
-export interface ListenerOptions {
-  host?: string;
-  port?: number;
-}
-
-/**
- * HTTP server options
- *
- * @export
- * @interface HttpOptions
- */
-export interface HttpOptions extends ListenerOptions {
-  protocol?: 'http';
-}
-
-/**
- * HTTPS server options
- *
- * @export
- * @interface HttpsOptions
- */
-export interface HttpsOptions extends ListenerOptions, https.ServerOptions {
-  protocol: 'https';
-}
-
-/**
- * Possible server options
- *
- * @export
- * @type HttpServerOptions
- */
-export type HttpServerOptions = HttpOptions | HttpsOptions;
-
-/**
- * Supported protocols
- *
- * @export
- * @type HttpProtocol
- */
-export type HttpProtocol = 'http' | 'https'; // Will be extended to `http2` in the future
+import {
+  HttpProtocol,
+  HttpServer,
+  HttpServerOptions,
+  RequestListener,
+} from './types';
 
 /**
  * HTTP / HTTPS server used by LoopBack's RestServer
  *
  * @export
  * @class HttpServer
  */
-export class HttpServer {
+export class DefaultHttpServer implements HttpServer {
```

Comment:
What's the impact of this change on possible users of `@loopback/http-server`?

For example, the following code creates a fully working HttpServer class in the current version (before your changes):

```ts
import * as turbo from 'turbo-http';

class TurboHttpServer extends HttpServer {
  constructor(
    requestListener: RequestListener,
    serverOptions?: HttpServerOptions,
  ) {
    super(requestListener, serverOptions);
    this.server = turbo.createServer(requestListener);
  }
}
```

IMO, we should preserve backwards compatibility in our API. Otherwise a new major version must be published, plus we will have to keep maintaining the old version according to our LTS policy...

Reply:
We can release `@loopback/http-server@2.0.0` if we have to break the compatibility due to the nature of this work.

Reply:
> We can release @loopback/http-server@2.0.0 if we have to break the compatibility due to the nature of this work.

Fair enough. In that case:

- The commit message in this PR should clearly describe the breaking changes so that our release tooling can pick it up. Ideally, there should be a short migration guide too.
- We need to update http-server's README with a table listing supported major versions (LTS terms). See https://github.com/strongloop/loopback#module-long-term-support-policy for inspiration. This should be done *before* 2.0.0 release, so that the new content is displayed on https://www.npmjs.com/package/@loopback/http-server

Reply:
See also https://github.com/strongloop/loopback-next/pull/2103